evm/src/interpreter.rs

```diff
@@ -934,32 +899,50 @@ impl Interpreter {
                         let r = H256::from(self.stack.pop());
                         topics.push(r);
                     }
-                    let data = self
-                        .mem
-                        .get(mem_offset.low_u64() as usize, mem_len.low_u64() as usize);
+                    let data = self.mem.get(mem_offset.low_u64() as usize, mem_len.low_u64() as usize);
                     self.logs.push(Log(topics, Vec::from(data)));
                 }
                 opcodes::OpCode::CREATE | opcodes::OpCode::CREATE2 => {
+                    // clear return data buffer before creating new call frame.
+                    self.return_data = vec![];
+
                     let value = self.stack.pop();
                     let mem_offset = self.stack.pop();
                     let mem_len = self.stack.pop();
-                    let salt = {
+                    let salt = H256::from({
                         match op {
                             opcodes::OpCode::CREATE => U256::zero(),
                             opcodes::OpCode::CREATE2 => self.stack.pop(),
                             _ => panic!("instruction can only be CREATE/CREATE2 checked above"),
                         }
-                    };
-                    let data = self
-                        .mem
-                        .get(mem_offset.low_u64() as usize, mem_len.low_u64() as usize);
-
+                    });
+                    let data = self.mem.get(mem_offset.low_u64() as usize, mem_len.low_u64() as usize);
+                    if value > self.data_provider.get_balance(&self.params.address) {
+                        self.stack.push(U256::zero());
+                        self.gas += self.gas_tmp;
+                        continue;
+                    }
```

Comment:
I think you have forgotten to remove the above code.

Reply:
A mistake, emmmm